astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,289 @@
+TCTYP_RE_ALGO = re.compile(r'(?P<algo>[A-Z]+)\s*')
+
+# FITS Time standard specified time units
+FITS_TIME_UNIT = ['s', 'd', 'a', 'cy', 'min', 'h', 'yr', 'ta', 'Ba']
+
+# Global time reference coordinate keywords
+TIME_KEYWORDS = {'TIMESYS' : 'scale', 'MJDREF' : 'ref_mjd',
+                 'JDREF' : 'ref_jd', 'DATEREF' : 'ref_date',
+                 'TREFPOS' : 'pos', 'TREFDIR' : 'dir',
+                 'TIMEUNIT' : 'unit', 'TIMEOFFS' : 'offs',
+                 'OBSGEO-X' : 'loc_x', 'OBSGEO-Y' : 'loc_y',
+                 'OBSGEO-Z' : 'loc_z', 'DATE' : 'date',
+                 'DATE-OBS' : 'date-obs', 'DATE-AVG' : 'date-avg',
+                 'DATE-BEG' : 'date-beg', 'DATE-END' : 'date-end',
+                 'MJD-OBS' : 'mjd-obs', 'MJD-AVG' : 'mjd-avg',
+                 'MJD-BEG' : 'mjd-beg', 'MJD-END' : 'mjd-end'}
+
+# Set astropy time global information
+GLOBAL_TIME_INFO = {'TIMESYS' : ('UTC','Default time scale'),
+                    'JDREF' : (0.0,'Time columns are jd = jd1 + jd2'),
+                    'TREFPOS' : ('TOPOCENTER','Time reference position')}
+
+
+def _verify_time_info(global_info, time_columns):
+    """
+    Given the time coordinate keyword information, verify that
+    each keyword has a valid value.
+    Also verify that a coordinate column of another type is not
+    mistaken to be time.
+    """
+    # FITS deprecated scales
+    global_info['scale'] = FITS_DEPRECATED_SCALES.get(global_info['scale'],
+                                                      global_info['scale'].lower())
+
+    # Verify global time scale
+    if not global_info['scale'] in Time.SCALES:
+        raise AssertionError(
+            'Global time scale (TIMESYS) must have a FITS recognized '
+            'time scale value (got {0!r}).'.format(global_info['scale']))
+
+    for idx, time_col in time_columns.items():
+        scale = time_col.get('scale', None)
+        unit = time_col.get('unit', None)
+
+        if scale is not None:
+            if scale.lower() in Time.SCALES:
+                time_col['scale'] = scale.lower()
+            elif scale in FITS_DEPRECATED_SCALES.keys():
+                time_col['scale'] = FITS_DEPRECATED_SCALES[scale]
+            elif scale == 'TIME':
+                time_col['scale'] = global_info['scale']
+            else:
+                if scale in ['GPS', 'LOCAL']:
+                    warnings.warn(
+                        'Table column {} has a FITS recognized time scale value {}. '
+                        'However, since it is not a valid astropy time scale, '
+                        'it will not be converted to astropy Time.'.format(idx, scale),
+                        AstropyUserWarning)
+                # Non-linear coordinate types have "4-3" form and are not time coordinates
+                elif TCTYP_RE_TYPE.match(scale[:5]) and TCTYP_RE_ALGO.match(scale[5:]):
+                    pass
+                else:
+                    warnings.warn(
```

Comment:
Does this have the potential to generate spurious warnings for valid non-Time type columns? Not sure, but we don't want to spam users in this case.